pkg/controller/config/config_controller.go

```diff
@@ -249,14 +251,22 @@ func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result
 
 	// If the watch set has not changed, we're done here.
 	if r.watched.Equals(newSyncOnly) && r.processExcluder.Equals(newExcluder) {
-		return reconcile.Result{}, nil
+		if !r.dirty {
+			return reconcile.Result{}, nil
+		}
+	} else {
+		// only re-calculate replays if our watch set has changed
+		r.needsReplay = nil
 	}
+	r.dirty = true
 
 	// --- Start watching the new set ---
 
 	// This must happen first - signals to the opa client in the sync controller
 	// to drop events from no-longer-watched resources that may be in its queue.
-	needReplay := r.watched.Union(newSyncOnly)
+	if r.needsReplay != nil {
```

Comment:
Is this condition backwards? Should needsReplay be recalculated one when it _is_ empty?

Reply:
It is totally backwards, thanks for catching that.